From d077770b9acf0128bf8d3f02a37e5ce4d15883d2 Mon Sep 17 00:00:00 2001
From: Pavel Tikhomirov <ptikhomirov@virtuozzo.com>
Date: Thu, 23 Dec 2021 17:19:49 +0300
Subject: [PATCH 207/245] crtools: move check_options after kerndat_init and
 log_init

This can be useful to check options which depend on some kernel features
listed in kdat.

Signed-off-by: Pavel Tikhomirov <ptikhomirov@virtuozzo.com>
---
 criu/crtools.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/criu/crtools.c b/criu/crtools.c
index 3c3491603..86aeb4d2b 100644
--- a/criu/crtools.c
+++ b/criu/crtools.c
@@ -185,9 +185,6 @@ int main(int argc, char *argv[], char *envp[])
 		return cr_service_work(atoi(argv[optind + 1]));
 	}
 
-	if (check_options())
-		return 1;
-
 	if (opts.imgs_dir == NULL)
 		SET_CHAR_OPTS(imgs_dir, ".");
 
@@ -264,6 +261,9 @@ int main(int argc, char *argv[], char *envp[])
 		return 1;
 	}
 
+	if (check_options())
+		return 1;
+
 	if (fault_injected(FI_CANNOT_MAP_VDSO))
 		kdat.can_map_vdso = 0;
 
-- 
2.35.1

